tasks.py

```diff
@@ -0,0 +1,142 @@
+    print('Updating release version in setup.cfg')
+    setupcfg_path = os.path.join(REPO_PATH, 'setup.cfg')
+    config = RawConfigParser()
+    config.read(setupcfg_path)
+
+    release_config = get_config(config)
+    if release_config['github_private'] and 'GITHUB_TOKEN' not in os.environ.keys():
+        print('\n'.join(textwrap.wrap(
+            'In order to grab pull request information from a private repository '
+            'you will need to set the GITHUB_TOKEN env variable with a personal '
+            'access token from GitHub.'
+        )))
+        return False
+
+    # Set the release number
+    config.set('metadata', 'version', version)
+    with open(setupcfg_path, 'wb') as configfile:
+        config.write(configfile)
+
+    # Install and run
+    print('Installing github-changelog')
+    ctx.run('npm install git+https://github.com/agjohnson/github-changelog.git')
+    if not since:
+        # Get last modified date from Git instead of assuming the file metadata is
+        # correct. This can change depending on git reset, etc.
+        git_log = ctx.run('git log -1 --format="%ad" -- CHANGELOG.rst')
+        since = parse(git_log.stdout.strip()).strftime('%Y-%m-%d')
+    changelog_path = os.path.join(REPO_PATH, 'CHANGELOG.rst')
+    template_path = os.path.join(
+        REPO_PATH,
+        'common',
+        'changelog.hbs',
+    )
+    bin_path = os.path.join(REPO_PATH, 'node_modules', '.bin')
+    cmd = (
+        '{bin_path}/gh-changelog '
+        '-o {owner} -r {repository} '
+        '--file {changelog_path} '
+        '--since {since} '
+        '--template {template_path} '
+        '--header "Version {version}" '
+        '--merged '
+    ).format(
+        bin_path=bin_path,
+        owner=release_config['github_owner'],
+        repository=release_config['github_repo'],
+        version=version,
+        template_path=template_path,
+        changelog_path=changelog_path,
+        since=since,
+    )  # yapf: disable
+    try:
+        token = os.environ['GITHUB_TOKEN']
+        cmd += '--token ' + token + ' '
+    except KeyError:
+        print('')
+        print(
+            '\n'.join(
+                textwrap.wrap(
+                    'In order to avoid rate limiting on the GitHub API, you can specify '
+                    'an environment variable `GITHUB_TOKEN` with a personal access token. '
+                    'There is no need for the token to have any permissions unless the '
+                    'repoistory is private.')))
+        print('')
+    print('Updating changelog')
+    ctx.run(cmd)
+
+
+@task
+def release(ctx, version):
+    """
+    Tag release of Read the Docs.
+
+    Do this after prepare task and manual cleanup/commit
+    """
+    # Ensure we're on the master branch first
+    git_rev_parse = ctx.run('git rev-parse --abbrev-ref HEAD', hide=True)
+    current_branch = git_rev_parse.stdout.strip()
+    if current_branch != 'master':
+        print('You must be on master branch!')
```

Comment:
What about adding something like

```python
input("""
Hey! Did you already:
1. ran `invoke prepare`
2. organize CHANGELOG.rst removing invalid PRs
3. cleanup commits
4. etc...
? [y/N]""")
```

to avoid mistakes and remind what are the proper steps to be followed.

Reply:
This would make a good enhancement. I think we can automate the checks though:

* Check setup.cfg for a matching version to the one you're releasing
* Check the changelog for a version string matching the one you're releasing
* Check that setupcfg and changelog changes are commited